### PR TITLE
Use ESLint Webpack plugin instead of eslint-loader

### PR DESCRIPTION
**Pull request by wmarques**
`eslint-loader` is deprecated so we should switch to the ESLint Webpack plugin which should give us better perfs
---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/package.json.ejs [hunk 1]
```diff
@@ -104,7 +104,7 @@ limitations under the License.
     "css-loader": "3.5.3",
     "eslint": "6.8.0",
     "eslint-config-prettier": "6.11.0",
-    "eslint-loader": "4.0.2",
+    "eslint-webpack-plugin": "2.1.0",
```
Comment (kaidohallik) [inline generators/client/templates/react/package.json.ejs:107]: React builds are failing. Probable reason: `eslint-webpack-plugin` version 2 minimal supported ESLint version is 7 but React is still on version 6.

Comment (wmarques) [reply generators/client/templates/react/package.json.ejs:107]: Is the upgrade easy ? I can try to upgrade if you want

Comment (DanielFran) [reply generators/client/templates/react/package.json.ejs:107]: @wmarques Better to separate into 2 PRs?

Comment (kaidohallik) [reply generators/client/templates/react/package.json.ejs:107]: For Angular I choosed this strategy for ESLint upgrade:
* update ESLint dependencies + turn off all violated rules in the latest ESLint dependencies
* rule by rule try to resolve errors

ATM 4 PR-s done for that: #12828, #12835, #12845, #12889

I agree that ESLint dependencies update in React should be a separate PR (or PR-s).

Another option for migrating to `eslint-webpack-plugin` in React side is to use version 1.0.0 which should be ESLint 6 compatible.

**Comment by DanielFran** [comment]
@wmarques Can you fix the conflicts?

File: generators/client/templates/angular/package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "browser-sync-webpack-plugin": "2.2.2",
     "codelyzer": "6.0.1",
     "copy-webpack-plugin": "6.2.1",
-    "eslint-loader": "4.0.2",
+    "eslint-webpack-plugin": "2.1.0",
```
Comment (DanielFran) [inline generators/client/templates/angular/package.json:27]: Update version to 2.2.0

File: generators/client/templates/react/package.json.ejs (hunk 1 above)
Comment (wmarques) [reply generators/client/templates/react/package.json.ejs:107]: Ok I'm trying to upgrade and I'll see how it goes :) If there is too much I'll stick with the 1.0.0

Comment (wmarques) [reply generators/client/templates/react/package.json.ejs]: @kaidohallik I successfully upgraded eslint related deps by switching some rules to off unfortunately :(

**Comment by wmarques** [comment]
@DanielFran @kaidohallik hope it's good :)

**Review by mshima — APPROVED** [review]
LGTM

**Review by DanielFran — APPROVED** [review]

**Review by DanielFran — APPROVED** [review]
LGTM

**Comment by mshima** [comment]
Missing another rule:
`@typescript-eslint/no-floating-promises` at https://github.com/hipster-labs/jhipster-daily-builds/runs/1415626195?check_suite_focus=true

**Pull request merged**